compiler/optimizer/InductionVariable.cpp

```diff
@@ -6382,6 +6382,27 @@ TR_InductionVariableAnalysis::analyzeExitEdges(TR_RegionStructure *loop,
 
       if (subNode->getStructure()->asRegion())
          {
+         // OSR blocks can be ignored here as they will not be impacted by a loop controlling
+         // condition. Exit edges do not have structure for their destination. Instead, find
+         // the corresponding block as a predecessor of the CFG exit based on the number.
+         TR::Block *dest = NULL;
+         int32_t toNum = edge->getTo()->getNumber();
+         TR::CFGNode *end = comp()->getFlowGraph()->getEnd();
+         for (auto predEdge = end->getPredecessors().begin(); predEdge != end->getPredecessors().end(); ++predEdge)
```

Comment:
This seems an expensive way of checking and are we sure it will always find the right exit block? You are using the exit of the CFG to find single level predecessors? I don't follow how that will always work...